ivy/functional/frontends/paddle/fft.py

```diff
@@ -131,3 +131,48 @@ def rfftfreq(n, d=1.0, dtype=None, name=None):
     pos_max = n // 2 + 1
     indices = ivy.arange(0, pos_max, dtype=dtype)
     return indices * val
+
+
+@with_supported_dtypes(
+    {"2.5.1 and below": ("complex64", "complex128")},
+    "paddle",
+)
+@to_ivy_arrays_and_back
+def irfftn(x, s=None, axes=None, norm="backward", name=None):
+    x = ivy.array(x)
+    
+    # Determine the output data type based on the input data type
+    if ivy.is_complex_dtype(x.dtype):
+        output_dtype = "float32" if x.dtype == "complex64" else "float64"
+    else:
+        output_dtype = "float32"
+        
+    target_axes = axes
+    
+    if s is None:
+        s = [x.shape[axis] if axis != axes[-1] else 2 * (x.shape[axis] - 1) for axis in axes]
```

Comment:
I tried running your function against the native version in the paddle docs:
```
import paddle
from ivy.functional.frontends import paddle as ivy_paddle

x = paddle.to_tensor([2.+2.j, 2.+2.j, 3.+3.j]).astype(paddle.complex128)

irfftn_x = paddle.fft.irfftn(x)
print(irfftn_x)

irfftn_x = ivy_paddle.irfftn(x)
print(irfftn_x)
```

Your implementation returns an error that stems from this line

Reply:
hi @mobley-trent, i think i need to add  this to handle when axes is not given

if axes is None:
        axes = list(range(len(x.shape)))

Reply:
I have update the function to handle mutiple axes case, the key is to isolately reverse each axis in the case, and return them at final step.

Reply:
@mobley-trent can we just call backend function from paddle, or other backend in frontend instead of using ivy libaray to develop the irfftn?

Reply:
@mobley-trent Hi Eddy, in my old version the bug is it doesn't handle the none axes case, and for mutiple axes, it does not update the local x, which is relative to freq domain if when it goes to mutiple axes.

In this updated irfftn, it will handle the none axes case, it either evidences that the irfftn is working for many axes case,
so rather than test natively which cost extra efforts, i think now it should be okay unless we want to irfftn has ability to cut off or extend the shape of output freq domain which is discussed earlier why s is set to none

Reply:
> @mobley-trent can we just call backend function from paddle, or other backend in frontend instead of using ivy libaray to develop the irfftn?

Most frontends are just a simple call to a similar ivy function, or a composition of other ivy functions. This serves as an abstraction to the backends. You are not supposed to implement a frontend function with a native function.

Reply:
yea but currently there is no irfftn in ivy backend to handle the case, it is more likely to use ifftn to simulate the irfftn, but for initial purpose the irfftn support for n dim, with just one axis or many specific axes, do you think this PR needs to be further developed?

Reply:
so @mobley-trent Eddy which one do you perfer
1. add a irfftn backend function in ivy\functional\ivy\experimental\layers.py, then this serve the purpose of abstract at frontend
2. keep things does not change like now

Reply:
@mobley-trent 
![image](https://github.com/unifyai/ivy/assets/87846736/ae821758-d9dc-4951-a758-2f0c4de15aa2)
if test natively, please differectly define function, then when generete a tensor, please directly copy the array into irfftn
becasue the frontend of ivy will return an x tensor, we pass the x[0] which is the array to our function and backend

x = paddle.to_tensor([[2.+2.j, 2.+2.j, 3.+3.j], [2.+2.j, 2.+2.j, 3.+3.j]]).astype(paddle.complex128)
- ivy_test_result=irfftn([[2.+2.j, 2.+2.j, 3.+3.j], [2.+2.j, 2.+2.j, 3.+3.j]])
- paddle_test_result=paddle.fft.irfftn(x)